File: backend/src/main/java/com/smithlife/backend/entity/Membership.java
package com.smithlife.backend.entity;

import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;

import java.time.LocalDate;
import java.time.LocalDateTime;

@Entity
@Table(name = "membership")
@Getter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Membership {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "membership_id")
    private Long membershipId;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", nullable = false)
    private User user;

    @Column(name = "type", nullable = false, length = 50)
    private String type;

    @Column(name = "start_date", nullable = false)
    private LocalDate startDate;

    @Column(name = "end_date", nullable = false)
    private LocalDate endDate;

    @Enumerated(EnumType.STRING)
    @Column(name = "status", nullable = false)
    private Status status;

    @CreationTimestamp
    @Column(name = "created_at", nullable = false, updatable = false)
    private LocalDateTime createdAt;

    public enum Status {
        ACTIVE, EXPIRED, PAUSED, CANCELLED
    }
}

File: frontend/lib/data/models/membership/membership.dart
class Membership {
  final int membershipId;
  final String type;
  final String startDate;
  final String endDate;
  final String status;
  final int remainingDays;

  Membership({
    required this.membershipId,
    required this.type,
    required this.startDate,
    required this.endDate,
    required this.status,
    required this.remainingDays,
  });

  factory Membership.fromJson(Map<String, dynamic> json) {
    return Membership(
      membershipId: json['membershipId'],
      type: json['type'],
      startDate: json['startDate'],
      endDate: json['endDate'],
      status: json['status'],
      remainingDays: json['remainingDays'] ?? 0,
    );
  }
}
